src/native/libs/System.Native/CMakeLists.txt

```diff
@@ -41,6 +41,15 @@ else()
     )
 endif()
 
+if (CLR_CMAKE_TARGET_APPLE OR CLR_CMAKE_TARGET_ANDROID OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
+    list (APPEND NATIVE_SOURCES
+        pal_crossprocessmutex_unsupported.c)
```

Comment:
We don't create a separate file in libs.native for unsupported targets. Instead, we use inline conditional directive to return -1 or ENOTSUP etc.